src/integrations/stripe.ts

```diff
@@ -162,6 +143,86 @@ export async function setupPaymentMethodAndPay(
   return null;
 }
 
+/**
+ * Used to either collect payment directly or launch stripe checkout to collect payment.
+ */
+export async function collectPayment(
+  actionData: ActionData,
+  stripePromise = getStripe()
+) {
+  const {
+    client,
+    triggerElement, // action button that triggered this
+    formattedFields,
+    updateFieldValues,
+    integrationData: stripeConfig
+  } = actionData;
+
+  // sync fields to BE
+  await syncFieldChanges(client, formattedFields, stripeConfig);
```

Comment:
this is fine for now i think but eventually these backend calls would ideally be unified so the data update is atomic and we save an extra network call

Reply:
No changes made rn.